src/api/environment.cc

```diff
@@ -87,12 +87,30 @@ static void HostCleanupFinalizationGroupCallback(
 }
 
 void* NodeArrayBufferAllocator::Allocate(size_t size) {
+  total_mem_usage_ += size;
   if (zero_fill_field_ || per_process::cli_options->zero_fill_all_buffers)
     return UncheckedCalloc(size);
```

Comment:
`UncheckedCalloc` allocates `size*sizeof(void*)` and also uses size=1 if size is 0:
https://github.com/nodejs/node/blob/e66a2acc4cb9fc09fc32d1833b89ae56468a0931/src/util-inl.h#L364-L369

Does `total_mem_usage` need to take into account the `sizeof` multiplier and the `size==0` case? What about the unlikely event that `calloc` fails?

Reply:
> Does `total_mem_usage` need to take into account the `sizeof` multiplier 

The multiplier is `1` in this case here, because `UncheckedCalloc()` without an explicit template type defaults to `char` (and `sizeof(char) == 1` always):

https://github.com/nodejs/node/blob/e66a2acc4cb9fc09fc32d1833b89ae56468a0931/src/util-inl.h#L396

> and the `size==0` case?

I think it would be odd to account for our (also odd) implementation detail of always returning at least a length-one allocation – I’d count that as overhead of the memory allocation implementation, just like the overhead that `malloc()` and friends add by themselves.

With the current solution, the counter always remains consistent with the actual `ArrayBuffer` allocations.

> What about the unlikely event that `calloc` fails?

I think V8 is always going to handle that by crashing, but I’ve updated the PR to also handle those situations gracefully – PTAL